### ISSUE TITLE
[Security] DOS vulnerability in ws module required by engine.io module

### ISSUE THREAD

**Issue by jdforsythe** [socketio/socket.io#3109]
*Note*: for support questions, please use one of these channels: [stackoverflow](http://stackoverflow.com/questions/tagged/socket.io) or [slack](https://socketio.slack.com)

### You want to:

* [x] report a *bug*
* [ ] request a *feature*

### Current behaviour

Dependency downstream on `ws` module

https://github.com/socketio/engine.io/issues/542
https://nodesecurity.io/advisories/550

### Steps to reproduce (if the current behaviour is a bug)

**Note**: the best way to get a quick answer is to provide a failing test case, by forking the following [fiddle](https://github.com/darrachequesne/socket.io-fiddle) for example.

### Setup
- OS:
- browser:
- socket.io version: 2.0.4

### Other information (e.g. stacktraces, related issues, suggestions how to fix)

Update to a version of socketio/engine.io with socketio/engine.io#542 fixed

**Comment by hello2dj** [comment]
#### like this
<img width="762" alt="socket io" src="https://user-images.githubusercontent.com/10645383/32645875-16aaa004-c5af-11e7-87e4-da44a97643dc.png">

**Comment by JREAM** [comment]
temporary work-around since this is 2 days old from posting.
touch `.nsprc`
```
{
  "exceptions":  ["https://nodesecurity.io/advisories/550"]
}
```

**Comment by carpii** [comment]
For anyone (like me) who wasnt even aware of nsp...

It's worth mentioning this workaround is only to stop nsp from reporting it 

It's not a workaround for the actual vulnerability.

**Comment by darrachequesne** [comment]
Closed by https://github.com/socketio/engine.io/pull/543 and released as [3.1.4](https://github.com/socketio/engine.io/releases/tag/3.1.4). Sorry for the delay!
